components/camera/rtppassthrough/stream_subscription.go

```diff
@@ -64,9 +60,8 @@ func NewStreamSubscription(queueSize int, onError func(error)) (*StreamSubscript
 	}
 
 	return &StreamSubscription{
-		buffer:  buffer,
-		id:      uuid.New(),
-		onError: onError,
+		buffer: buffer,
+		id:     uuid.New(),
```

Comment:
First time I'm really looking at this. Is this StreamSubscription uuid the same as what gets put into `Subscription.Id`? There's a 1-1 mapping from StreamSubscription object to Subscription object?

Reply:
1. Yes `Subscription.Id` is a `StreamSubscription` uuid
2. Yes, but keep in mind that in the module case there will be 2 `Subscription`s, one in the camera/client.go and one in the module's camera.

Reply:
> one in the camera/client.go and one in the module's camera.

I'm a bit confused at this. Why are there?

Reply:
![Screenshot 2024-05-01 at 3 26 02 PM](https://github.com/viamrobotics/rdk/assets/5927876/9e7035f7-bf16-4835-92d8-b4b819284104)
Does this help?

Reply:
Yes helpful :)